### DISCUSSION TITLE
Minix S912 - auto boot after power outage

### DISCUSSION THREAD

**Discussion by dm-imdy** [devmfc/debian-on-amlogic#72] [Q&A] [answered]
Hi,
Last week I've installed Ubuntu from your latest release (v6.6.34) on my Minix neo U9-H (S912).
It was easy as all I had to do is to uncomment `box=s912_generic_gigabit` in the `boot.config` file.
Thank you!

My setup:
1. Ubuntu is installed on sd card (keeping the original Android TV v6 untouched)
2. Executed `/root/aml-multiboot-setup.sh` to enable switching boot between stock firmware and Ubuntu

I was looking for a way that the box will boot automatically when the power returns after power outage, but I couldn't find such.
Eventually, I solved it by:
1. Pulled out the sd card and booted the box to the original firmware (Android TV v6)
2. Went to `Settings->Advanced Settings->MCU setting` and turn this option.
3. Reinserted the sd card and booted to Ubuntu
and...voila... when unplugging and re-plugging the power supply, the box boots automatically to Ubuntu!

My question:
Is there a way to achieve the same, without booting to the stock firmware?

Comment:
Seems like a bug in `/root/aml-multiboot-setup.sh`. There is a redundant `)` in the line:
`fw_setenv cmd_boot_org "$(fw_printenv) -n bootcmd)"  || exit_with_message "Error backupping original bootcmd"`
It took me some time to understand why multi boot didn't work. It did work after this small fix:
`fw_setenv cmd_boot_org "$(fw_printenv -n bootcmd)"  || exit_with_message "Error backupping original bootcmd"`

Thanks.

**Comment by devmfc** [comment, marked as answer]
I think that is something very specific to the Minix bootloader, because I have never seen that behaviour in one of my boxes. They always just boot when I plug in the power.
So I can't help you with the first question.

Argh, where did that come from.... Thanks for catching that! It will be fixed in the next release.
